core/jvm/test/PoolingTest.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2024 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io
+
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+class PoolingTest {
+    @Test
+    fun segmentSharing() {
```

Comment:
Should we also add tests for using Byte and writing one buffer to another?
*and Buffer.copy

Reply:
There's already a test case covering the `.copy()`, if I got you right.
Added a case with `.write`.